packages/debug/src/browser/debug-session.tsx

```diff
@@ -221,6 +224,9 @@ export class DebugSession implements CompositeTreeElement {
         this.fireDidChange();
         if (thread) {
             this.toDisposeOnCurrentThread.push(thread.onDidChanged(() => this.fireDidChange()));
+
+            // If this thread is missing stack frame information, then load that.
+            this.updateFrames();
```

Comment:
Why do we need to update frames whenever current thread is changed?

Reply:
Like the comment says,

> If this thread is missing stack frame information, then load that.

On a breakpoint or other stop, we clear all threads' frames and only load the current thread's stackframes. When you switch thread it's going to be empty. Which is good, because fetching all stackframes on stop would be slow on a large number of threads.

Reply:
I meant that we should update only according to `stop` logic, if we do it probably then by the time when someone changes a thread frames will be already there or at least updating. This code should not be necessary.